### PR TITLE
Convert hashicorp/waypoint to GitHub Actions

### PR DESCRIPTION
**Pull request by hc-github-team-es-release-engineering**
Pipeline migrated from [CircleCI](https://app.circleci.com/pipelines/github/hashicorp/waypoint) :tada:

## Manual steps

Perform the following steps to complete the migration:

### hashicorp/waypoint/go-tests
- [x] Ensure environment variable is updated: `ALGOLIA_API_KEY`
- [x] Ensure environment variable is updated: `GH_DOCKER_PASS`
- [x] Ensure environment variable is updated: `GH_DOCKER_USER`
- [x] Ensure environment variable is updated: `SLACK_APPROVAL_NOTIFICATION_WEBHOOK_URL`
- [x] Ensure environment variable is updated: `SLACK_WEBHOOK`

### hashicorp/waypoint/integration
- [x] ~Ensure secret is available: `${{ secrets.CIRCLE_CI_SSH_KEY }}`~
- [x] ~Ensure secret is available: `${{ secrets.CIRCLE_CI_KNOWN_HOSTS }}`~
- [x] ~Ensure environment variable is updated: `ALGOLIA_API_KEY`~
- [x] ~Ensure environment variable is updated: `GH_DOCKER_PASS`~
- [x] ~Ensure environment variable is updated: `GH_DOCKER_USER`~
- [x] ~Ensure environment variable is updated: `SLACK_APPROVAL_NOTIFICATION_WEBHOOK_URL`~
- [x] ~Ensure environment variable is updated: `SLACK_WEBHOOK`~

### hashicorp/waypoint/integration-hcl
- [x] ~Ensure environment variable is updated: `ALGOLIA_API_KEY`~
- [x] ~Ensure environment variable is updated: `GH_DOCKER_PASS`~
- [x] ~Ensure environment variable is updated: `GH_DOCKER_USER`~
- [x] ~Ensure environment variable is updated: `SLACK_APPROVAL_NOTIFICATION_WEBHOOK_URL`~
- [x] ~Ensure environment variable is updated: `SLACK_WEBHOOK`~

### hashicorp/waypoint/website-mdx
- [x] ~~Ensure environment variable is updated: `ALGOLIA_API_KEY`~~
- [x] ~~Ensure environment variable is updated: `GH_DOCKER_PASS`~~
- [x] ~~Ensure environment variable is updated: `GH_DOCKER_USER`~~
- [x] ~~Ensure environment variable is updated: `SLACK_APPROVAL_NOTIFICATION_WEBHOOK_URL`~~
- [x] ~~Ensure environment variable is updated: `SLACK_WEBHOOK`~~

### hashicorp/waypoint/json-docs
- [x] ~~Ensure environment variable is updated: `ALGOLIA_API_KEY`~~
- [x] ~~Ensure environment variable is updated: `GH_DOCKER_PASS`~~
- [x] ~~Ensure environment variable is updated: `GH_DOCKER_USER`~~
- [x] ~~Ensure environment variable is updated: `SLACK_APPROVAL_NOTIFICATION_WEBHOOK_URL`~~
- [x] ~~Ensure environment variable is updated: `SLACK_WEBHOOK`~~

### hashicorp/waypoint/frontend
- [x] Ensure environment variable is updated: `ALGOLIA_API_KEY`
- [x] Ensure environment variable is updated: `GH_DOCKER_PASS`
- [x] Ensure environment variable is updated: `GH_DOCKER_USER`
- [x] Ensure environment variable is updated: `SLACK_APPROVAL_NOTIFICATION_WEBHOOK_URL`
- [x] Ensure environment variable is updated: `SLACK_WEBHOOK`

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+          timeout 60s bash -c 'trap "echo Failed, timeout reached while waiting for kubernetes to launch 1>&2" SIGTERM; while ! kubectl get node 2>/dev/null ; do sleep 1 ; done; echo "Connected, ok"'
+      - uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        with:
+          name: waypoint.tar
+      - name: Execute Integration Tests
+        run: |-
+          tar -xvf waypoint.tar
+          ./ci/integration.sh
+
+# FIXME: Is this CRT-able?
```
Comment (dekimsey) [inline .github/workflows/integration.yml:123]: > FIXME: Is this CRT-able?

We are currently working on seeing if we can move this test to use the images CRT produces in `build.yml`.

File: .github/workflows/go-tests.yml [hunk 2]
```diff
@@ -0,0 +1,142 @@
+name: go-tests
+on:
+  pull_request:
+  push:
+
+jobs:
+  # FIXME: We already lint in integration.yml, which should we keep?
```
Comment (dekimsey) [inline .github/workflows/go-tests.yml:7]: > FIXME: We already lint in integration.yml, do we need to be linting twice? Which should we keep? 

I'm thinking we can remove the one from the `integration.yml` and keep this which is doing unit testing anyway.

**Review by dekimsey — CHANGES_REQUESTED** [review]
Highlighted some of the FIXME/discussion points.

File: .github/workflows/integration.yml (hunk 1 above)
Comment (dekimsey) [reply .github/workflows/integration.yml:123]: We successfully ported this code to the CRT build.yml.

File: .github/workflows/go-tests.yml (hunk 2 above)
Comment (dekimsey) [reply .github/workflows/go-tests.yml]: I removed the integration.yml linting in favor of this workflow

**Review by dekimsey — APPROVED** [review]

**Review by catsby — APPROVED** [review]
👍 Thank you so much!

**Pull request merged**